backend/src/main/java/ca/bc/gov/backendstartapi/entity/seedlot/Seedlot.java

```diff
@@ -0,0 +1,203 @@
+package ca.bc.gov.backendstartapi.entity.seedlot;
+
+import ca.bc.gov.backendstartapi.entity.embeddable.AuditInformation;
+import ca.bc.gov.backendstartapi.enums.GeneticClassEnum;
+import ca.bc.gov.backendstartapi.enums.SeedlotStatusEnum;
+import jakarta.persistence.Column;
+import jakarta.persistence.Embedded;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.io.Serial;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import lombok.Generated;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+/** A registered seedlot. */
+@Entity
+@Table(name = "seedlot")
+@NoArgsConstructor
+@RequiredArgsConstructor
+@Getter
+@Setter
+public class Seedlot implements Serializable {
+
+  @Generated @Serial private static final long serialVersionUID = -8731389710757531855L;
+
+  @Id
+  @Column(name = "seedlot_number", length = 5)
+  @NonNull
+  private String id;
+
+  @Column(name = "seedlot_status_code", length = 3, nullable = false)
+  @NonNull
```

Comment:
They're there for different purposes. `nullable = false` is for the JPA, to make sure it won't update this column with a null value; `@NonNull` is there to prevent Lombok destructive methods to assign a null value to the object's property.

I used `@NonNull` because we cannot commit a seedlot with a null status to the database and we don't have a reason to allow an entity without it for now (which may change in the future).

`nullable = false` should be there anyway, because it prevents us from wasting a call to the database with data that can't be committed. And while we should be safe with the `@NonNull`, I don't trust we'll be free of reflection shenanigans that might bypass our encapsulation and assign a null value to the status.

About not using `nullable = false` in the identifier, it was because `@Id` already takes care of it. But now that I think about it, I think there's no reason not to use it.